templates/daemonset.yaml

```diff
@@ -36,39 +36,51 @@ spec:
         {{- if $enableCore }}
         - name: core
           args: [ core ]
-          image: {{ printf "%s:%s" .Values.agent.image.repository .Values.agent.image.tag }}
+          image: "{{ .Values.agent.image.repository }}:{{ include "datafy-agent.agentImageTag" . }}"
           imagePullPolicy: {{ .Values.agent.image.pullPolicy }}
           {{- with .Values.agent.securityContext }}
           securityContext:
             {{- toYaml . | nindent 12 }}
           {{- end }}
           envFrom:
             - configMapRef:
-                name: datafy-agent-envvars
+                name: {{ include "datafy-agent.fullname" . }}-cm
           env:
             - name: DATAFY_DSO_URL
               value: {{ .Values.agent.dsoUrl }}
             - name: DATAFY_TOKEN
               valueFrom:
                 secretKeyRef:
-                  name: {{ default "datafy-token" .Values.agent.externalTokenSecret.name }}
+                  name: {{ default (printf "%s-token" (include "datafy-agent.fullname" .)) .Values.agent.externalTokenSecret.name }}
                   key: {{ default "token" .Values.agent.externalTokenSecret.key }}
           resources:
             requests:
               cpu: 10m
               memory: 40Mi
+          {{- if .Values.agent.readinessProbe.enabled }}
+          readinessProbe:
+            exec:
+              command:
+                - pgrep
+                - -f
+                - datafy-agent
```

Comment:
what is this probe? 
1. this is the core container, so `datafy-agent` is not running here.
2. the `core` is the main process of the container, if its down, the container will restart